ops/lib.rs

```diff
@@ -0,0 +1,250 @@
+            }
+          } else {
+            default_ok
+          }
+        }
+        _ => default_ok,
+      };
+
+      quote! {
+        match result {
+          Ok(v) => {
+            #ok_block
+          },
+          Err(err) => {
+            let err = deno_core::serde_v8::to_v8(
+              scope,
+              deno_core::OpError {
+                class_name: (op_state.get_error_class_fn)(&err),
+                message: err.to_string(),
+                code: deno_core::error_codes::get_error_code(&err),
+              },
+            ).unwrap();
+
+            rv.set(err);
+          },
+        };
+      }
+    }
+  };
+
+  let is_async = func.sig.asyncness.is_some();
+  if is_async {
+    TokenStream::from(quote! {
+      #[allow(non_camel_case_types)]
+      pub struct #name;
+
+      impl #name {
+        pub fn name() -> &'static str {
+          stringify!(#name)
+        }
+
+        pub fn v8_cb #generics () -> deno_core::v8::FunctionCallback #where_clause {
+          use deno_core::v8::MapFnTo;
+          Self::v8_func::<#type_params>.map_fn_to()
+        }
+
+        pub fn decl #generics ()  -> (&'static str, deno_core::v8::FunctionCallback) #where_clause {
+          (Self::name(), Self::v8_cb::<#type_params>())
+        }
+
+        #original_func
+
+        pub fn v8_func #generics (
+          scope: &mut deno_core::v8::HandleScope,
+          args: deno_core::v8::FunctionCallbackArguments,
+          mut rv: deno_core::v8::ReturnValue,
+        ) #where_clause {
+          use deno_core::JsRuntime;
+          use deno_core::futures::FutureExt;
+          use deno_core::OpCall;
+          use deno_core::to_op_result;
+          use deno_core::PromiseId;
+          use deno_core::bindings::throw_type_error;
+          use deno_core::v8;
+          let op_id = unsafe { v8::Local::<v8::Integer>::cast(args.get(0)) }.value() as usize;
+
+          let promise_id = args.get(1);
+          let promise_id = v8::Local::<v8::Integer>::try_from(promise_id)
+            .map(|l| l.value() as PromiseId)
+            .map_err(deno_core::anyhow::Error::from);
+          // Fail if promise id invalid (not an int)
+          let promise_id: PromiseId = match promise_id {
+            Ok(promise_id) => promise_id,
+            Err(err) => {
+              throw_type_error(scope, format!("invalid promise id: {}", err));
+              return;
+            }
+          };
+
+          #a
+          #b
+          #func
+
+          let state_rc = JsRuntime::state(scope);
+          let mut state = state_rc.borrow_mut();
+          state.op_state.borrow().tracker.track_async(op_id);
+          state.have_unpolled_ops = true;
+          let op_state = state.op_state.clone();
+          let op_state2 = op_state.clone();
+          state.pending_ops.push(OpCall::eager(async move {
+            let result = #name::<#type_params>(op_state.clone(), a, b).await;
+            (promise_id, op_id, to_op_result(&op_state.borrow(), result))
```

Comment:
Hmm, would it make sense to, again with the same `()` return value condition, use a specific `to_null_op_result` function that could then (presumably) avoid serializing the Promise resolve value?